src/storage/store/var_list_column.cpp

```diff
@@ -12,12 +12,24 @@ namespace kuzu {
 namespace storage {
 
 offset_t ListOffsetInfoInStorage::getListOffset(uint64_t nodePos) const {
-    if (nodePos == 0) {
-        return prevNodeListOffset;
-    } else {
-        auto offsetVector = offsetVectors[(nodePos - 1) / DEFAULT_VECTOR_CAPACITY].get();
-        return offsetVector->getValue<offset_t>((nodePos - 1) % DEFAULT_VECTOR_CAPACITY);
-    }
+    return getEndListOffset(nodePos) - getListLength(nodePos);
+}
+
+offset_t ListOffsetInfoInStorage::getEndListOffset(uint64_t nodePos) const {
+    if (offsetVectors.size() == 0)
+        return offset_t(0);
+    if (nodePos == numTotal)
+        return getEndListOffset(nodePos - 1);
```

Comment:
I don't think these two if conditions make sense.
For the first one, I don't think we should allow accessing under this case `offsetVectors.size() == 0`;
For the second one, should also not happen.